pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -184,15 +209,25 @@ let
       setuptoolsBuildHook
     ] ++ lib.optionals (format == "flit") [
       flitBuildHook
-    ] ++ lib.optionals (format == "pyproject") [
-      pipBuildHook
-    ] ++ lib.optionals (format == "wheel") [
+    ] ++ lib.optionals (format == "pyproject") [(
+      if python.isPy2 then
```

Comment:
The builder should not bother anymore with Python 2. For Python 2 there can be a copy of the builder, probably just the legacy one using pip.